specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/examples/Disks/Get.json

```diff
@@ -0,0 +1,27 @@
+{
+  "parameters": {
+    "subscriptionId": "04666444-56f0-4d4f-afc5-dbd6b134b084",
+    "location": "redmond",
+    "DiskId": "423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+    "api-version": "2018-07-30-preview"
+  },
+    "responses": {
+        "200": {
+            "body": {
+                "id": "/subscriptions/04666444-56f0-4d4f-afc5-dbd6b134b084/providers/Microsoft.Compute.Admin/locations/redmond/disks/423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+                "name": "redmond/423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+                "type": "Microsoft.Compute.Admin/locations/disks",
+                "location": "redmond",
+              "properties": {
+                "diskId": "423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+                "status": "Unattached",
+                "sharePath": "\\\\SU1FileServer.mydomain.com\\SU1_ObjStore_4",
```

Comment:
this is very not REST/HTTP friendly. Do these file shares have corresponding REST resources who's IDs could be referenced here instead (maybe FRP?). Resources should reference other resources by their IDs.

Reply:
We need to integrate with FRP to fetch the resource ID. currently diskrp doesn't do the integration. It is great to have. I will open a work item to track what we can do.

Reply:
Can you share the ID-generation logic? Sharing a library might be not that hard. I assume this change won't be a breaking change, so delaying might be OK if you are under pressure.

Reply:
I think you mean the file share? There is no file share generation logic.  
There is another command in SRP which will list all file shares and it contains the share path. User could call that command to get the share out and then call DRP to pass the sharepath in. 

Integrate with FRP, I mean, in the long term, SRP might not need to maintain the share resource. and DRP can check with FRP what's the possible share path is.